File: 1/main.py
#!/usr/bin/env python3
import sys
from pprint import pprint

def solve_file(file_name):
    print(f"Solving file: {file_name}")
    with open(file_name, "r") as infile:
        numbers = [ int(i) for i in infile.readlines() ]
        product = [ i*j for i in numbers for j in numbers if i + j == 2020 ][0]
        print(f"Product: {product}")

def main():
    list(map(solve_file, sys.argv[1:]))

main()

File: 1/main.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature 'say';

sub solve_file($) {
  my $filename = shift;
  say "Solving for file $filename";
  open my $infile < $filename or die "Bad file $filename: $!";
  my @lines = <$infile>;
  close $infile;
  my @numbers = map {scalar($_)} @lines;
  for my $i (@numbers) {
    for my $j (@numbers) {
      say $i * $j and return if $i + $j eq 2020;
    }
  }
}

solve_file($_) for @ARGV;
